File: selfstudy/golang/transactions_min.go
/*
task 233:
Transactions - Min
You need to find the minimum value among transaction amounts in dollars
*/

package main

import (
	"fmt"
	"reflect"
)

func LowestTransactions(amounts []int) int {
	if len(amounts) == 0 { return 0 }
	m := amounts[0]
	for _, v := range amounts {
		if v < m { m = v }
	}
	return m
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, LowestTransactions([]int{10, 20, 30})},
		{[]int{50, 60, 70}, LowestTransactions([]int{50, 60, 70})},
		{[]int{1, 2, 3}, LowestTransactions([]int{1, 2, 3})},
		{[]int{100, 200}, LowestTransactions([]int{100, 200})},
		{[]int{}, LowestTransactions([]int{})},
	}
	
	for idx, t := range tests {
		res := LowestTransactions(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done

File: selfstudy/js/transactions_min.js
/*
task 33:
Transactions - Min
You need to find the minimum value among transaction amounts in dollars
*/

function lowest_transactions(amounts) {
  return amounts.length ? Math.min(...amounts) : 0;
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: lowest_transactions([10, 20, 30]) },
  { input: [50, 60, 70], expected: lowest_transactions([50, 60, 70]) },
  { input: [1, 2, 3], expected: lowest_transactions([1, 2, 3]) },
  { input: [100, 200], expected: lowest_transactions([100, 200]) },
  { input: [], expected: lowest_transactions([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('lowest_transactions', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(lowest_transactions(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, lowest_transactions(t.input) === t.expected || JSON.stringify(lowest_transactions(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// finally works
